bin/dump-to-git.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+
+var fs = require('fs');
+var process = require('process');
+var child_process = require('child_process');
+var Promise = require('bluebird');
+var NodeGit = require('nodegit');
+var LZString = require('lz-string');
+var models = require("./lib/models");
+
+var repoRootDir = './notes';
+
+process.chdir(repoRootDir);
+
+
+function exportNotes() {
+  return models.Note
+  .findAll()
+  .then(function (notes) {
+    // Iterate over all notes
+    return Promise.all(notes.map(function (note) {
+      var file = LZString.compressToBase64(note.id);
+
+      // Export note to file
+      return new Promise(function (resolve) {
+        console.log('Exporting ' + note.id + ' to ' + file);
+        var body = LZString.decompressFromBase64(note.content);
+        fs.writeFile(file, body, resolve);
+      })
+      .then(function () {
+        return file;
+      });
+
+    }));
+  });
+}
+
+
+function addToIndexAndReturnTree(files, index) {
+  return Promise.all(files.map(function (file) {
+    console.log('Adding ' + file + ' to git index');
+    return index.addByPath(file);
+  }))
+  .then(function () {
+    console.log('Write files to git index');
+    return index.write();
+  })
+  .then(function () {
+    console.log('Write git index to tree');
+    return index.writeTree();
+  });
+}
+
+
+function getHeadCommit(repo) {
+  return NodeGit.Reference
+  .nameToId(repo, 'HEAD')
+  .then(function (headRef) {
+    return repo.getCommit(headRef);
+  });
+}
+
+
+function commit(repo, tree) {
+  return getHeadCommit(repo)
+  .then(function (head) {
+    console.log('Commiting...');
+    var sig = NodeGit.Signature.now('Committer', 'johndoe@localhost');
```

Comment:
can nodegit take the regular configuration from git itself? (`git config user.name` / `git config user.email`)
If it can, i'd leave this part out.
If it can't this should definitely be configurable.